modules/contact/src/auxkernels/MortarArchardsLawAux.C

```diff
@@ -0,0 +1,128 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "MortarArchardsLawAux.h"
+#include "MooseVariableFE.h"
+#include "FEProblemBase.h"
+#include "Assembly.h"
+
+registerMooseObject("ContactApp", MortarArchardsLawAux);
+
+InputParameters
+MortarArchardsLawAux::validParams()
+{
+  InputParameters params = MortarNodalAuxKernel::validParams();
+  params.addClassDescription(
+      "Returns the weighted gap velocity at a node. This quantity is useful for mortar contact, "
+      "particularly when dual basis functions are used in contact mechanics");
+  params.addCoupledVar("v",
+                       "Optional variable to take the value of. If omitted the value of the "
+                       "`variable` itself is returned.");
+  params.addRequiredCoupledVar("disp_x", "The x displacement variable");
+  params.addRequiredCoupledVar("disp_y", "The y displacement variable");
+  params.addCoupledVar("disp_z", "The z displacement variable");
```

Comment:
I think we've been moving away from this pattern for a while, instead using `displacements` to couple multiple components (using `coupledValues` into a `std::vector<VariableValue *>`

Reply:
We need `MooseVariables` to have access to methods to obtain velocities and neighbor velocities, don't we?